File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        sub = []
        def dfs(i):
            if i >= len(nums):
                ans.append(sub.copy())
                return
            
            #decision to include nums[i]
            sub.append(nums[i])
            dfs(i+1) #will include a subset including nums[i]

            #decision not to include nums[i]
            sub.pop()
            dfs(i+1)#subset wont include nums[i]

        dfs(0)
        return ans
            
            
            
         

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {

        int val = 0;
        for(int i =0;i<nums.length-1;i++){
            val=nums[i];
            for(int j = i+1;j<nums.length;j++){
                if(nums[j]==val){
                    return true;
                }
            }
        }
        return false;
    }
}
